app/controllers/api/v0/manifests_controller.rb

```diff
@@ -12,30 +8,31 @@ def search
   end
 
   def validate
-    begin
-      manifest_content = JSON.parse(request.body.read)
+    manifest_content = read_body_as_json
+    unless performed?
```

Comment:
`performed?` is just avoiding multi-render bug. http://apidock.com/rails/ActionController/Metal/performed%3F

Since we have consolidated the error handling, it's possible we have already rendered an error before we get to the success render.